src/neo-vm/UnhandledException.cs

```diff
@@ -0,0 +1,28 @@
+using Neo.VM.Types;
+using System;
+using System.Text;
+using Array = Neo.VM.Types.Array;
+
+namespace Neo.VM
+{
+    public class UnhandledException : Exception
+    {
+        public StackItem ExceptionObject { get; }
+
+        public UnhandledException(StackItem e) : base(GetExceptionMessage(e))
+        {
+            ExceptionObject = e;
+        }
+
+        private static string GetExceptionMessage(StackItem e)
+        {
+            StringBuilder sb = new StringBuilder("An unhandled exception was thrown.");
+            if (e is Array array && array.Count > 0 && array[0] is ByteString s)
```

Comment:
What if e is not an array? As of preview 3, it looks like the stack item that gets thrown is a bytestring

Reply:
If `e` is not an array, the default message will be returned.

Reply:
Throwing `System.Exception` with a single string parameter does not create an array of VM parameters. I suspect this will be the most common scenario for unhandled VM exceptions. There seems little point to making this change if this scenario is not supported.

This C#:
``` csharp
public static int GetAnswer()
{
    throw new Exception("Earth blown up by Vogon constructor fleet");
}
```

is converted into this .NET IL
``` 
.method public hidebysig static 
	int32 GetAnswer () cil managed 
{
	.maxstack 8

	IL_0000: nop
	IL_0001: ldstr "Earth blown up by Vogon constructor fleet"
	IL_0006: newobj instance void [netstandard]System.Exception::.ctor(string)
	IL_000b: throw
} 
```

Which in turn is converted to this NeoVM code by NEON 3 preview 3
```
# Start Method DeepThought.getAnswer
000 NOP
001 PUSHDATA1 45-61-72-74-68-20-62-6C-6F-77-6E-20-75-70-20-62-79-20-56-6F-67-6F-6E-20-63-6F-6E-73-74-72-75-63-74-6F-72-20-66-6C-65-65-74 # as text: "Earth blown up by Vogon constructor fleet"
044 NOP
045 THROW
# End Method DeepThought.getAnswer
```

Reply:
Updated.